runtime/lua/vim/_core/server.lua

```diff
@@ -0,0 +1,26 @@
+local M = {}
+
+--- Helper function for serverlist(), this returns all the currently running servers
+--- in stdpath("run"). Does not include named pipes or TCP servers.
+---
+--- @return string[] A list of currently running servers in stdpath("run")
+function M.serverlist()
+  local socket_paths = vim.fs.find(function(name, _)
```

Comment:
not a blocker for your work, but let's mention a TODO here.
```suggestion
  -- TODO: also get named pipes on Windows
  local socket_paths = vim.fs.find(function(name, _)
```